addons/html_builder/static/src/builder/plugins/common_between_options.js

```diff
@@ -0,0 +1,74 @@
+import { Plugin } from "@html_editor/plugin";
+import { isMobileView } from "@html_builder/builder/utils/utils";
+
+export class CommonBetweenOptions extends Plugin {
+    static id = "common-between-options";
+    resources = {
+        option_visibility_update: optionVisibilityUpdate,
+        clean_for_save_handlers: this.cleanForSave.bind(this),
+    };
+    static shared = ["toggleTargetVisibility"];
+
+    setup() {
+        this.builderOptions = this.getResource("builder_options");
+    }
+
+    cleanForSave({ root }) {
+        for (const option of this.builderOptions) {
+            const { selector, exclude, clean_for_save_handlers_options } = option;
+            let editingEls = [...root.querySelectorAll(selector)];
+            if (exclude) {
+                editingEls = editingEls.filter((editingEl) => !editingEl.matches(exclude));
+            }
+            editingEls.map((editingEl) => {
+                const show =
+                    !editingEl.classList.contains("o_snippet_invisible") &&
+                    !editingEl.classList.contains("o_snippet_mobile_invisible") &&
+                    !editingEl.classList.contains("o_snippet_desktop_invisible");
+                this.toggleTargetVisibility(editingEl, show);
+                if (clean_for_save_handlers_options) {
+                    clean_for_save_handlers_options(editingEl);
+                }
+            });
+        }
+    }
+
+    toggleTargetVisibility(editingEl, show) {
+        show = optionVisibilityUpdate({ editingEl: editingEl, show: show });
+        const dispatchName = show ? "target_show" : "target_hide";
+        this.dispatchTo(dispatchName, editingEl);
+        return show;
+    }
```

Comment:
In order to be more clear/less confusing in general, maybe call it `show` when it will show  and `isShown` after the action
Like here, you want to show so the argument `show` in toggleTargetVisibility is okay
But when you call `optionVisibilityUpdate`, the result is wheter it was shown or not, so here I would declare a `isShown` variable instead of reusing `show` :slightly_smiling_face: 

And then, do the same in `invisible_elements_panel.js`